packages/session-replay-browser/src/helpers.ts

```diff
@@ -1,13 +1,102 @@
 import { getGlobalScope } from '@amplitude/analytics-client-common';
-import { UNMASK_TEXT_CLASS } from './constants';
+import { MASK_TEXT_CLASS, UNMASK_TEXT_CLASS } from './constants';
+import { DEFAULT_MASK_LEVEL, MaskLevel, PrivacyConfig } from './config/types';
+import { getInputType } from '@amplitude/rrweb-snapshot';
 
-export const maskInputFn = (text: string, element: HTMLElement) => {
-  if (element.classList?.contains(UNMASK_TEXT_CLASS)) {
-    return text;
+// compiler guard
+/* istanbul ignore next */
+const assertUnreachable = (_: never): never => {
+  throw new Error('did not expect to get here');
+};
+
+/**
+ * Light: Subset of inputs
+ * Medium: All inputs
+ * Conservative: All inputs and all texts
+ */
+const isMaskedForLevel = (element: HTMLElement, elementType: 'input' | 'text', level: MaskLevel): boolean => {
+  switch (level) {
+    case MaskLevel.LIGHT: {
+      if (elementType !== 'input') {
+        return false;
+      }
+
+      const inputType = getInputType(element);
+      /* istanbul ignore if */ // For some reason it's impossible to test this.
+      if (!inputType) {
+        return false;
+      }
+
+      if (['password', 'hidden', 'email', 'tel'].includes(inputType)) {
+        return true;
+      }
+
+      if ((element as HTMLInputElement).autocomplete.startsWith('cc-')) {
+        return true;
+      }
+
+      return false;
+    }
+    case MaskLevel.MEDIUM:
+      return elementType === 'input';
+    case MaskLevel.CONSERVATIVE:
+      return elementType === 'input' || elementType === 'text';
   }
-  return '*'.repeat(text.length);
+  /* istanbul ignore next */
+  return assertUnreachable(level);
 };
 
+/**
+ * Checks if the given element set to be masked by rrweb
+ *
+ * Priority is:
+ *  1. [In code] Element/class based masking/unmasking
+ *  2. [Config based] Selector based masking/unmasking
+ *  3. Use app defaults
+ */
+const isMasked = (element: HTMLElement, elementType: 'input' | 'text', config?: PrivacyConfig): boolean => {
+  // Element is explicitly instrumented to mask/unmask
+  if (element.classList) {
+    if (element.classList.contains(MASK_TEXT_CLASS)) {
+      return true;
+    } else if (element.classList.contains(UNMASK_TEXT_CLASS)) {
+      return false;
+    }
+  }
+
+  // No privacy config? Default to medium policy
+  if (!config) {
+    return isMaskedForLevel(element, elementType, DEFAULT_MASK_LEVEL);
+  }
+
+  // Config has override for mask/unmask
+  const shouldMask = (config.maskSelector ?? []).some((selector) => element.matches(selector));
+  if (shouldMask) {
+    return true;
+  }
+  const shouldInclude = (config.includeSelector ?? []).some((selector) => element.matches(selector));
```

Comment:
maybe we should use "unmask" terminology instead of include here? since in the UI exclude translates to block, this adds more confusion.

Reply:
Did you mean in the variable name or the config field?

Reply:
probably good to update both.